frontend/public/components/secrets/create-secret.tsx

```diff
@@ -0,0 +1,155 @@
+        }
+        history.push(formatNamespacedRouteForResource('secrets'));
+      },
+      err => this.setState({error: err.message, inProgress: false})
+    );
+  }
+  render () {
+    const title = `${this.props.titleVerb} ${_.upperFirst(this.state.secretType)} Secret`;
+    const { saveButtonText } = this.props;
+
+    let explanation = 'Webhook secret allow you to authenticate a webhook trigger.';
+    let subform = <WebHookSecretSubform onChange={this.onDataChanged.bind(this)} stringData={this.state.stringData} />;
+
+    return <div className="co-m-pane__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form className="co-m-pane__body-group" onSubmit={this.save}>
+        <h1 className="co-m-pane__heading">{title}</h1>
+        <p className="co-m-pane__explanation">{explanation}</p>
+
+        <fieldset disabled={!this.props.isCreate}>
+          <div className="form-group">
+            <label className="control-label">Secret Name</label>
+            <div>
+              <input className="form-control" type="text" onChange={this.onNameChanged} value={this.state.secret.metadata.name} required id="test--subject-name" />
+              <p className="help-block">Unique name of the new secret.</p>
+            </div>
+          </div>
+        </fieldset>
+        {subform}
+        <ButtonBar errorMessage={this.state.error} inProgress={this.state.inProgress} >
+          <button type="submit" className="btn btn-primary" id="create-secret">{saveButtonText || 'Create'}</button>
+          <Link to={formatNamespacedRouteForResource('secrets')} className="btn btn-default">Cancel</Link>
+        </ButtonBar>
+      </form>
+    </div>;
+  }
+}
+
+const BaseEditSecret = connect(null, {setActiveNamespace: UIActions.setActiveNamespace})(
+  (props: BaseEditSecretProps_) => <BaseEditSecret_ {...props} />
+);
+
+const BindingLoadingWrapper = props => {
+  const fixed = _.reduce(props.fixedKeys, (acc, k) => ({...acc, k: _.get(props.obj.data, k)}), {});
+  return <StatusBox {...props.obj}>
+    <BaseEditSecret {...props} obj={props.obj.data} fixed={fixed} />
+  </StatusBox>;
+};
+
+export const CreateSecret = ({match: {params}}) => {
+  return <BaseEditSecret
+    fixed={{ metadata: {namespace: params.ns} }}
+    metadata={{ namespace: getActiveNamespace() }}
+    secretType={params.type}
+    titleVerb="Create"
+    isCreate={true}
+  />;
+};
+
+export const EditSecret = ({match: {params}, kind}) => <Firehose resources={[{kind: kind, name: params.name, namespace: params.ns, isList: false, prop: 'obj'}]}>
+  <BindingLoadingWrapper fixedKeys={['kind', 'metadata']} titleVerb="Edit" saveButtonText="Save Changes" />
+</Firehose>;
+
+/* eslint-disable no-undef */
+export type BaseEditSecretState_ = {
+  secretType?: string,
+  secret: K8sResourceKind,
+  inProgress: boolean,
+  type: string,
+  stringData: {[key: string]: string},
+  error?: any,
+};
+
+export type BaseEditSecretProps_ = {
```

Comment:
Can we eliminate the `any` types?
For example:
```
 obj: K8sResourceKind 
// ...
metadata: {
  name: string,
  // etc.
}
```
I think [this file](https://github.com/alecmerdler/console/blob/13ec0f68fe5a8f37403b67585aad1fc1b2ab89a4/frontend/public/components/cloud-services/index.tsx) has some better type arbitrary examples.

Reply:
`fixed` can really be any... It's just saying always use these properties. I don't think we can define that one.

We can use K8sResourceKind for obj

Reply:
> Can we eliminate the any types?

no all the types have to be typed, otherwise yarn will screen, eg:
```
Property 'metadata' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<Pick<BaseEditSecretProps_, "fixed" | "ki...'
```

Reply:
Since some parts of codebase are not typed the its almost impossible not to use `any`